tests/testthat/test_prepareAnnotations.R

```diff
@@ -1,37 +1,150 @@
 context("Prepare annotations")
 
-test_that("prepareAnnotations of txdb object is a GRangesList", {
+test_that("prepareAnnotations of txdb object (seqnames, ranges, strand) matches the expectation", {
     txdb <- AnnotationDbi::loadDb(system.file("extdata", "Homo_sapiens.GRCh38.91.annotations-txdb_chr9_1_1000000.sqlite", package = "bambu"))
+    
+    expectedGR <- readRDS(system.file("extdata", "annotationGranges_txdb2Grch38_91_chr9_1_1000000.rds", package = "bambu"))
+    
+    gr <- prepareAnnotations(x = txdb)
+    
+    expect_equal(lapply(gr, granges), lapply(expectedGR, granges))
+})
+
 
+test_that("prepareAnnotations of a path to gtf file (seqnames, ranges, strand) matches the expectation", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+    
     expectedGR <- readRDS(system.file("extdata", "annotationGranges_txdbGrch38_91_chr9_1_1000000.rds", package = "bambu"))
+    
+    gr <- prepareAnnotations(x = gtf.file)
+    
+    expect_equal(lapply(gr, granges), lapply(expectedGR, granges))
+})
+
 
+test_that("prepareAnnotations of txdb object (metadata) matches the expectation", {
+    txdb <- AnnotationDbi::loadDb(system.file("extdata", "Homo_sapiens.GRCh38.91.annotations-txdb_chr9_1_1000000.sqlite", package = "bambu"))
+  
+    expectedGR <- readRDS(system.file("extdata", "annotationGranges_txdb2Grch38_91_chr9_1_1000000.rds", package = "bambu"))
+  
     gr <- prepareAnnotations(x = txdb)
+    
+    expect_named(mcols(gr), c("TXNAME", "GENEID", "txid", "eqClass", "eqClassById"))
+    expect_equal(mcols(gr), mcols(expectedGR))
+})
 
-    expect_equal(gr, expectedGR)
-    expect_s4_class(gr, class = "CompressedGRangesList")
-    expect_named(mcols(gr), c("TXNAME", "GENEID", "eqClass"))
+
+test_that("prepareAnnotations of a path to gtf file (metadata) matches the expectation", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+  
+    expectedGR <- readRDS(system.file("extdata", "annotationGranges_txdbGrch38_91_chr9_1_1000000.rds", package = "bambu"))
+  
+    gr <- prepareAnnotations(x = gtf.file)
+  
+    expect_named(mcols(gr), c("TXNAME", "GENEID", "txid", "eqClass", "eqClassById"))
+    expect_equal(mcols(gr), mcols(expectedGR))
 })
 
 
-test_that("prepareAnnotations of genome library is a GRangesList", {
-    library(TxDb.Hsapiens.UCSC.hg38.knownGene)
-    gr <- prepareAnnotations(TxDb.Hsapiens.UCSC.hg38.knownGene)
+test_that("positive strand gives ascending exon_rank", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+  
+    gr <- prepareAnnotations(x = gtf.file)
+  
+    positive_check <- data.frame(txname = names(unlist(gr)), unlist(gr)) %>% 
+        filter(strand == "+") %>% 
```

Comment:
- unlist(gr) called twice, probably could be done with unlist(gr) called only once
- can this be tested for negative strand as well? you could also add the test for exon end rank in here

Reply:
changed to call one time only